ts/parts/Chart.ts

```diff
@@ -1051,119 +1059,121 @@ extend(Chart.prototype, /** @lends Highcharts.Chart.prototype */ {
+            // Chart.setCaption
+            elem.update = function (
+                o: (Highcharts.DescriptionOptionsType)
+            ): void {
+                const fn = {
+                    title: 'setTitle',
+                    subtitle: 'setSubtitle',
+                    caption: 'setCaption'
+                }[name];
+                (chart as any)[fn](o);
+            };
 
-            if (title && titleOptions) {
-                (chart as any)[name] = title = title.destroy(); // remove old
+            // Presentational
+            if (!this.styledMode) {
+                elem.css((options as any).style);
             }
 
-            if (chartTitleOptions && !title) {
-                (chart as any)[name] = chart.renderer.text(
-                    chartTitleOptions.text as any,
-                    0,
-                    0,
-                    chartTitleOptions.useHTML
-                )
-                    .attr({
-                        align: chartTitleOptions.align,
-                        'class': 'highcharts-' + name,
-                        zIndex: (chartTitleOptions as any).zIndex || 4
-                    })
-                    .add();
+            /**
+             * The chart title. The title has an `update` method that allows
+             * modifying the options directly or indirectly via
+             * `chart.update`.
+             *
+             * @sample highcharts/members/title-update/
+             *         Updating titles
+             *
+             * @name Highcharts.Chart#title
+             * @type {Highcharts.TitleObject}
+             */
 
-                // Update methods, shortcut to Chart.setTitle
-                (chart as any)[name].update = function (
-                    o: (Highcharts.TitleOptions|Highcharts.SubtitleOptions)
-                ): void {
-                    chart.setTitle(
-                        (!i && o) as (Highcharts.TitleOptions|undefined),
-                        (i && o) as (Highcharts.SubtitleOptions|undefined)
-                    );
-                };
-
-                // Presentational
-                if (!styledMode) {
-                    (chart as any)[name].css(chartTitleOptions.style);
-                }
+            /**
+             * The chart subtitle. The subtitle has an `update` method that
+             * allows modifying the options directly or indirectly via
+             * `chart.update`.
+             *
+             * @name Highcharts.Chart#subtitle
+             * @type {Highcharts.SubtitleObject}
+             */
+            this[name] = elem;
+        }
 
-            }
-        });
-        chart.layOutTitles(redraw);
     },
 
     /**
-     * Internal function to lay out the chart titles and cache the full offset
-     * height for use in `getMargins`. The result is stored in
-     * `this.titleOffset`.
+     * Internal function to lay out the chart title, subtitle and caption and
```

Comment:
Commas missing: `Internal function to lay out the chart title, subtitle, and caption, and`

Reply:
Hmm. I don't think there should be a comma after "subtitle" (at least it's optional), but one after "caption" is probably needed.